### PR TITLE
feat(collections): fix get repositoryController on request

### PR DESCRIPTION
**Pull request by oscaryald**

### CODE REVIEW

File: packages/components/table/collections/xm-table-repository-collection-controller.service.ts [hunk 1]
```diff
@@ -126,6 +126,7 @@ export class XmTableRepositoryCollectionController<T = unknown>
         params?: QueryParams,
         headers?: HttpHeaders
     ): Observable<T> {
+        this.repository = this.repositoryController;
```
Comment (denisogolubev) [inline packages/components/table/collections/xm-table-repository-collection-controller.service.ts:129]: It looks like we do not follow DRY principles (4 times the same assign).

File: packages/components/table/collections/xm-table-repository-collection-controller.service.ts [hunk 2]
```diff
@@ -52,8 +52,8 @@ export class XmTableRepositoryCollectionController<T = unknown>
         super();
     }
 
-    private getRepositoryController(): IEntityCollectionPageable<T, PageableAndSortable> {
-        return this.xmDynamicInstanceService.getControllerByKey(
+    private get repositoryController(): IEntityCollectionPageable<T, PageableAndSortable> {
+        return this.repository || this.xmDynamicInstanceService.getControllerByKey(
```
Comment (denisogolubev) [inline packages/components/table/collections/xm-table-repository-collection-controller.service.ts:56]:
```ts
private getRepositoryController(): IEntityCollectionPageable<T, PageableAndSortable> {
        if (!this.repository) {
            this.repository = this.xmDynamicInstanceService.getControllerByKey(
                'repository',
                this.injector
            );
        }
        return this.repository;
    }
```
That will provide cache and you don't need to duplicate 4 times `this.repository = this.repositoryController;`. Just access `this.repositoryController.delete(id, params, headers) where you need.

File: packages/components/table/collections/xm-table-repository-collection-controller.service.ts (hunk 1 above)
Comment (oscaryald) [reply packages/components/table/collections/xm-table-repository-collection-controller.service.ts:129]: any ideas?

Comment (denisogolubev) [reply packages/components/table/collections/xm-table-repository-collection-controller.service.ts:129]: Please, look comment below.

Comment (oscaryald) [reply packages/components/table/collections/xm-table-repository-collection-controller.service.ts]: done

**Review by denisogolubev — APPROVED** [review]

**Review by LevBD — APPROVED** [review]

**Pull request merged**